service/s3/presign_post_test.go

```diff
@@ -0,0 +1,497 @@
+package s3
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/internal/awstesting/unit"
+	"github.com/aws/aws-sdk-go-v2/internal/sdk"
+)
+
+func TestPresignPutObject(t *testing.T) {
+	fixedTime := time.Date(2022, time.February, 1, 0, 0, 0, 0, time.UTC)
```

Comment:
Small tip, for testing like this if you find you're doing this temp-patch thing more often, this you can do something like

```go
func mockTime(t time.Time) func() {
    sdk.NowTime = func() time.Time { return t }
    return func() { sdk.NowTime = time.Now }
}

// in test
defer mockTime(time.Whatever)()
```

Reply:
Nice, changed